### ISSUE TITLE
Variable line width

### ISSUE THREAD

**Issue by MinooErho** [pgf-tikz/pgf#1289]
### Brief outline of the proposed feature

variable line width is absent in PGF/TiKZ.
Adobe Illustrator has this feature.

![photo_2023-10-31_23-31-12](https://github.com/pgf-tikz/pgf/assets/135517407/9d978a95-e361-4a7f-99b3-4954ebbd34f6)

**Comment by muzimuzhi** [comment]
Decoration opens some possibilities, see for example https://tex.stackexchange.com/questions/14283/stroke-with-variable-thickness.

Compared to Adobe Illustrator and some other GUI drawing softwares, PGF/TikZ line is just a simple wrapper around low-level move-to and line-to operations, followed by a stroke operation. The width, color, cap, etc. graphic states are all constant through out a PGF/TikZ line.

**Comment by MinooErho** [comment]
So it is a appropriate wish.

the link you sent is a simulation.

**Comment by hmenke** [comment]
There is no such thing as "variable line width" in the PDF spec. Adobe Illustrator is also just drawing a polygon. If you share the PDF I can point it out.

**Comment by MinooErho** [comment]
there must be such a command for simplicity.

**Comment by hmenke** [comment]
I don't believe there is, but here is a link to the PDF specification: https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/PDF32000_2008.pdf
Can you please point out where this command is described?

**Comment by MinooErho** [comment]
PGF should make such thing for user convenience, not PDF.

**Comment by hmenke** [comment]
But there is no such thing in PDF, so the best that PGF (and Adobe Illustrator for that matter) can do is a simulation using decorations which you explicitly rejected above. I don't see a way forward here.

**Comment by MinooErho** [comment]
misunderstood. :)

i don't mention to PDF format.
i mention to simulation.

no problem with simulation. but it is a lake that no macro for this drawing width.
for example see this curve in illustrator.

![photo_2023-10-31_23-31-12](https://github.com/pgf-tikz/pgf/assets/135517407/e932e9cd-1dc8-4f8a-99bf-007cbe850158)
